frontend/src/lib/components/Sharing/SharingModal.tsx

```diff
@@ -32,124 +33,144 @@ export function Sharing({ dashboardId, insightShortId, insight }: SharingModalPr
         whitelabelAvailable,
         sharingConfiguration,
         sharingConfigurationLoading,
+        showPreview,
         embedCode,
         iframeProperties,
         shareLink,
     } = useValues(sharingLogic(logicProps))
-    const { setIsEnabled } = useActions(sharingLogic(logicProps))
+    const { setIsEnabled, togglePreview } = useActions(sharingLogic(logicProps))
 
     const [iframeLoaded, setIframeLoaded] = useState(false)
 
-    const showNoLabelCheckbox = insight?.filters?.insight === InsightType.TRENDS
+    const showNoLegendCheckbox = insight?.filters?.insight === InsightType.TRENDS
     const resource = dashboardId ? 'dashboard' : 'insight'
 
     useEffect(() => {
         setIframeLoaded(false)
-    }, [iframeProperties.src, sharingConfiguration?.enabled])
+    }, [iframeProperties.src, sharingConfiguration?.enabled, showPreview])
```

Comment:
Do we also need the `noLegend` in here?

Reply:
Also not directly related to this change but I noticed if one would edit the Insight after having viewed the preview and then come back to the preview it shows the old iframe content.

Reply:
> Do we also need the noLegend in here?
Nope since the `src` already includes it.